test/nodes/test_web_retriever.py

```diff
@@ -27,6 +28,15 @@ def mocked_article_extractor():
         yield
 
 
+@pytest.fixture
+def mocked_link_content_fetcher_handler_type():
+    with patch(
+        "haystack.nodes.retriever.link_content.LinkContentFetcher._get_content_type_handler",
+        return_value=html_content_handler,
+    ):
+        yield
```

Comment:
I'm not sure why we need this patch. `_get_content_type_handler` is just looking up a dictionary, isn't it?

Reply:
It is but when running with mocks  the input for this method is a mock too and that blows up a few unit tests